nixos/modules/services/networking/dsnet.nix

```
@@ -0,0 +1,366 @@
+                      default = "";
+                      example = "c908j1sT3MIs4ac/mIwaiXn9/2CpI4HwcMX9WYXWUHc=";
+                      description = ''
+                        The preshared key of the peer/server combination
+                      '';
+                    };
+                  };
+                }
+              )
+            );
+
+            default = null;
+            description = ''
+              If you want to manage the peers from NixOS configuration, you can
+              add them here. This is not recommended as it negates the purpose
+              of dsnet in the first place.
+
+              If this is populated, like other values it will override any
+              existing peers.
+            '';
+          };
+
+          PersistentKeepalive = lib.mkOption {
+            type = lib.types.nullOr lib.types.int;
+            default = null;
+            example = 25;
+            description = ''
+              The persistent keepalive interval for the peer. This is used to
+              keep the connection alive when there is no traffic. This is
+              recommended to be set to 25 seconds.
+            '';
+          };
+        };
+      };
+
+      default = { };
+      description = ''
+        The settings to use for dsnet. This will be converted to a JSON
+        object that will be passed to dsnet as a patch, using the patch
+        command when the service is started. See the dsnet documentation for
+        more information on the available options.
+
+        Note that the resulting /etc/dsnetconfg.json is more of a database
+        than it is a configuration file. It is therefore recommended that
+        system specific values are configured here, rather than the full
+        configuration including peers.
+
+        Peers may be managed via the dsnet add/remove commands, negating the
+        need to manage key material and cumbersom configuration with nix. If
+        you want peer configuration in nix, you may as well use the regular
+        wireguard module.
+      '';
+      example = lib.literalExpression ''
+        {
+            ExternalHostname = "vpn.example.com";
+            ExternalIP = "127.0.0.1";
+            ExternalIP6 = "";
+            ListenPort = 51820;
+            Network = "10.3.148.0/22";
+            Network6 = "",
+            IP = "10.3.148.1";
+            IP6 = "";
+            DNS ="8.8.8.8";
+            Networks = ["0.0.0.0/0"];
+        }
+      '';
+    };
+  };
+
+  config = lib.mkIf cfg.enable {
+    environment.systemPackages = [ cfg.package ];
+
+    systemd.services.dsnet = {
+      description = "dsnet VPN Management";
+      after = [ "network-online.target" ];
+      wants = [ "network-online.target" ];
+      wantedBy = [ "multi-user.target" ];
+      preStart = ''
+        test ! -f /etc/dsnetconfig.json && ${cfg.package}/bin/dsnet init
+        cat ${patchFile} | ${cfg.package}/bin/dsnet patch
```

```suggestion
        test ! -f /etc/dsnetconfig.json && ${lib.getExe cfg.package} init
        cat ${patchFile} | ${lib.getExe cfg.package} patch
```
Sorry, forgot to mark here too!

---

Ah missed that. Thanks